Fix deprecation warnings

antoscarface opened the pull request:
The PR aims to solve some deprecation warnings complained by some users. Here the most important  ones:

```
Deprecated: Wordlift\Modules\Common\Symfony\Component\Config\Resource\ComposerResource implements the Serializable interface, which is deprecated. Implement __serialize() and __unserialize() instead (or in addition, if support for old PHP versions is necessary) in /var/www/html/wp-content/plugins/wordlift/modules/common/third-party/vendor/symfony/config/Resource/ComposerResource.php on line 18
  |  
  | Deprecated: Wordlift\Modules\Common\Symfony\Component\Config\Resource\FileResource implements the Serializable interface, which is deprecated. Implement __serialize() and __unserialize() instead (or in addition, if support for old PHP versions is necessary) in /var/www/html/wp-content/plugins/wordlift/modules/common/third-party/vendor/symfony/config/Resource/FileResource.php on line 20
  |  
  | Deprecated: Return type of Wordlift\Content\Wordpress\Wordpress_Content_Id::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/wp-content/plugins/wordlift/classes/content/wordpress/class-wordpress-content-id.php on line 63
```

The first twos come from an oldest Symfony libraries which needs to be updated, inside `modules/common`, which requires to have a min PHP req to at least 7.4+ in order to easily upgrade the library without putting checks about PHP version because of it needs different code to load base on server PHP version.

Also, I had to fix the bootstrap of PHP scoper autoload in order to include also the "autoload files" which are not supported by PHP scoper. This to solve some missing "native php functions" included by the polyfill libraries of symfony, which are required from the upgraded version of symfony dependencies.

In the PR are also fixes some other minor warnings.

The large edits in the PR comes from the dependencies update.

I tried to keep the commits speaking in order to go though them and undestand the flow of changes.

ziodave commented on src/modules/common/includes/vendor/composer/autoload_real.php:
```diff
@@ -23,11 +23,11 @@ public static function getLoader()
 
         spl_autoload_register(array('ComposerAutoloaderInit8bd2b6d4ccd6b50e0809e5ba62169420', 'loadClassLoader'), true, true);
-        self::$loader = $loader = new \Wordlift_Modules_Common_Composer\Autoload\ClassLoader(\dirname(__DIR__));
+        self::$loader = $loader = new \Composer\Autoload\ClassLoader(\dirname(__DIR__));
```
We need a custom scoped namespace or it can clash with other libraries.

ziodave commented on src/modules/common/includes/vendor/composer/autoload_real.php:
```diff
@@ -23,11 +23,11 @@ public static function getLoader()
 
         require __DIR__ . '/autoload_static.php';
-        call_user_func(\Wordlift_Modules_Common_Composer\Autoload\ComposerStaticInit8bd2b6d4ccd6b50e0809e5ba62169420::getInitializer($loader));
+        call_user_func(\Composer\Autoload\ComposerStaticInit8bd2b6d4ccd6b50e0809e5ba62169420::getInitializer($loader));
```
We need a custom scoped namespace or it can clash with other libraries.

ziodave commented on src/modules/common/includes/vendor/composer/autoload_static.php:
```diff
@@ -2,7 +2,7 @@
 // autoload_static.php @generated by Composer
 
-namespace Wordlift_Modules_Common_Composer\Autoload;
+namespace Composer\Autoload;
```
We need a custom scoped namespace or it can clash with other libraries.

antoscarface commented on src/libraries/action-scheduler/classes/schema/ActionScheduler_StoreSchema.php:
nope sorry I didn't noticed that, I gonna check for upgrading it instead of editing it. Do I just need to upgrade the package and replace the entire library manually correct? I'm not seeing a composer here to manage the dependency, if I'm not wrong.

ziodave replied:
Correct, if I remember correctly, it is just a drop-in.

antoscarface replied:
moved into a new commit which upgrade the library: https://github.com/insideout10/wordlift-plugin/pull/1746/commits/282e98524ddfa942d343e5e43b9c2b44b2a8733b

antoscarface commented:
@ziodave I fixed your comments (replied in one and for the others I made a [fixup](https://github.com/insideout10/wordlift-plugin/pull/1746/commits/ff8a265462df418e9aa35da3cef34b6b8e66101b) )

I've added also other fixes for other deprecation warnings I'm noticing it. 
I don't see any other warnings at the moment.

ziodave commented on src/libraries/action-scheduler/changelog.txt:
```diff
@@ -1,5 +1,103 @@
+= 3.8.2 - 2024-09-12 =
+* Add missing parameter to the `pre_as_enqueue_async_action` hook.
+* Bump minimum PHP version to 7.0.
+* Bump minimum WordPress version to 6.4.
```
I think we try to be compatible with WP 5.8+ ?

antoscarface replied:
right, used the version which fixed the warnings (just one minor up, `3.5.4 -> 3.5.3`), so no changes to wp min req
